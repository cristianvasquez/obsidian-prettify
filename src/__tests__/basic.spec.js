const prettifier = require('../prettifier')

test("Basic functionality: a formatted table", () => {
    const content = `
**A**|**B**|**C**
|---:|:---|---|
a |b |c
x |y |z
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Basic functionality: lists use * bullets", () => {
    const content = `
- a
- b
  - c
  - d
    - e
    - f
            `
    return prettifier(content, {bullet: '*'}).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Basic functionality: lists use - bullets", () => {
    const content = `
* a
* b
  * c
  * d
    * e
    * f
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Basic functionality: alternating keeps information of different codification", () => {
    const content = `
- a
- b
  * c
  - d
    - e
    * f
            `
    return prettifier(content, {bullet: '*'}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Basic functionality: enumerate lists", () => {
    const content = `
1. foo
  a. aaa
  b. bbb
  c. ccc
1. bar
1. baz
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Basic functionality: enumerate lists", () => {
    const content = `
1. foo
    1. aaa
    1. bbb
    4. ccc
1. bar
1. baz
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Basic functionality: dont break obsidian links", () => {
    const content = `
[[respect this]]
    [[and this]][[also this]]
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Embed youtube", () => {
    const content = `
https://www.youtube.com/watch?v=B6rKUf9DWRI
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Embed image", () => {
    const content = `
https://cyberculturesblog.files.wordpress.com/2019/01/Memex.jpg
            `
    return prettifier(content).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Indent size", () => {
    const content = `
- foo
-  foo
-   foo
-  foo
            `
    return prettifier(content,{listItemIndent:'one'}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Indent size", () => {
    const content = `
- foo
-  foo
-   foo
-  foo
            `
    return prettifier(content,{listItemIndent:'tab'}).then(data => {
        expect(data).toMatchSnapshot();
    });
});
