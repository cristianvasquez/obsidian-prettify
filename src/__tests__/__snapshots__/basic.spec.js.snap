// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic functionality: a formatted table 1`] = `
VFile {
  "contents": "| **A** | **B** | **C** |
| ----: | :---- | ----- |
|     a | b     | c     |
|     x | y     | z     |
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Basic functionality: alternating keeps information of different codification 1`] = `
VFile {
  "contents": "* a
* b
  * c

  <!---->

  * d
    * e

    <!---->

    * f
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Basic functionality: dont break obsidian links 1`] = `
VFile {
  "contents": "[[respect this]]
[[and this]][[also this]]
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Basic functionality: enumerate lists 1`] = `
VFile {
  "contents": "1. foo

a. aaa
b. bbb
c. ccc

1. bar
2. baz
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Basic functionality: enumerate lists 2`] = `
VFile {
  "contents": "1. foo
   1. aaa
   2. bbb
   3. ccc
2. bar
3. baz
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Basic functionality: lists use * bullets 1`] = `
VFile {
  "contents": "* a
* b
  * c
  * d
    * e
    * f
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Basic functionality: lists use - bullets 1`] = `
VFile {
  "contents": "- a
- b
  - c
  - d
    - e
    - f
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Embed image 1`] = `
VFile {
  "contents": "[![](https://cyberculturesblog.files.wordpress.com/2019/01/Memex.jpg)](https://cyberculturesblog.files.wordpress.com/2019/01/Memex.jpg)
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Embed youtube 1`] = `
VFile {
  "contents": "<https://www.youtube.com/watch?v=B6rKUf9DWRI>
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Indent size 1`] = `
VFile {
  "contents": "- foo
- foo
- foo
- foo
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Indent size 2`] = `
VFile {
  "contents": "-   foo
-   foo
-   foo
-   foo
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;
